doc/source/how-to/testing.rst

```diff
@@ -0,0 +1,536 @@
+
+        .. code-block:: text
+
+            python -m pip install .[test]
+
+    .. tab:: From requirements_tests.txt
+
+        .. code-block:: text
+
+            python -m pip install -r requirements_tests.txt
+
+
+Organizing Test Files
+~~~~~~~~~~~~~~~~~~~~~
+Test files must be collected in :ref:`The \`\`tests/\`\` Directory`. To
+guarantee that tests are run against the library source code, follow a ``src/``
+layout as explained in :ref:`The \`\`src/\`\` Directory` section rather than
+having your Python library source directly within the repository root directory. 
+
+This helps you avoid testing the source of the repository and
+rather the installed package. This helps to catch errors caused by files that
+might be missed by the installer, including any C extensions or additional
+internal packages.
+
+Running Tests
+-------------
+Once you have installed ``pytest``, it is possible to execute the test suite by
+running:
+
+.. code-block:: text
+
+    pytest -v tests/
+
+Filtering Tests
+~~~~~~~~~~~~~~~
+It is also possible to run a subset of all available tests by taking advantage
+of the ``keywords`` and ``markers`` flags:
+
+**Filtering tests by keywords**
+
+.. code-block:: text
+
+    pytest -k '<name pattern>'
+    pytest -k 'not <name pattern>'
+
+**Filtering tests by markers**
+
+.. code-block:: text
+
+    pytest -m slow
+
+
+Visit `Working with Custom Markers
+<https://docs.pytest.org/en/latest/example/markers.html>`_ for more information
+about filtering tests.
+
+Testing Methodology
+-------------------
+You should consider three levels of testing for your PyAnsys library: unit,
+integration, and functional testing.
+
+* :ref:`Unit Testing` validates your library at the lowest possible level, isolating
+  individual classes and methods without any communication with other libraries
+  or services.
+
+* :ref:`Integration Testing` validates that your library works in the context of an
+  application or software stack. For example, if your library extends or wraps
+  the functionality of an external service, you would need to test that service
+  in conjunction with your library. On GitHub, the ideal approach for this would
+  be to start your service via docker and test accordingly. You should still be
+  testing at the individual class or method level, but you can now test how
+  multiple libraries or services interact. This is mandatory for testing APIs and
+  is preferred over mocking the service.
+
+* :ref:`Functional Testing` should be used for validating workflows or long running
+  examples. For example, if you have a library that wraps a CAD service, you
+  would validate that you can create complex geometry while directly interfacing
+  with the service. Functional tests are great at discovering edge cases that are
+  not normally found at the unit or integration level, but functional testing
+  should be limited to only a handful of examples as these tend to be long
+  running and difficult to validate.
```

Comment:
```suggestion
  not normally found at the unit or integration level. However, functional testing
  should be limited to only a handful of examples because these tend to be long
  running and difficult to validate.
```